packages/mgt-components/src/components/mgt-login/mgt-login.ts

```diff
@@ -241,14 +241,11 @@ export class MgtLogin extends MgtTemplatedTaskComponent {
     }
 
     const provider = Providers.globalProvider;
-    if (provider?.isMultiAccountSupportedAndEnabled) {
-      localStorage.removeItem(provider.getActiveAccount().id + this._userDetailsKey);
-    }
     if (provider?.logout) {
       await provider.logout();
       this.userDetails = null;
       if (provider.isMultiAccountSupportedAndEnabled) {
-        localStorage.removeItem(provider.getActiveAccount().id + this._userDetailsKey);
+        localStorage.removeItem(provider.getActiveAccount()?.id + this._userDetailsKey);
```

Comment:
Now the item for the user who just got logged out is left behind as `provider.getActiveAccount()` can return a nullish value.

Suggest that the active account is read into a local variable before `provider.logout` is called that way there's an id to compose this key with. Calling localStorage.remove with a non-existent key doesn't cause an error, so the current code will run without throwing a runtime error but will not achieve the goals of this code.